src/test/java/com/unstoppabledomains/resolution/LibraryTest.java

```diff
@@ -0,0 +1,139 @@
+package com.unstoppabledomains.resolution;
+
+/*
+ * This Java source file was generated by the Gradle 'init' task.
+ */
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import com.unstoppabledomains.exceptions.NSExceptionCode;
+import com.unstoppabledomains.exceptions.NamingServiceException;
+
+import org.junit.jupiter.api.BeforeEach;
+
+public class LibraryTest {
+    Resolution resolution;
+
+    @BeforeEach
+    public void initEach() {
+        resolution = new Resolution("https://mainnet.infura.io/v3/781c1e5cae32417b93eac26042950d25");
+    }
+
+    @Test
+    public void namehash() {
+        try {
+            final String hash = resolution.namehash("crypto");
+            assertEquals("0x0f4a10a4f46c288cea365fcf45cccf0e9d901b945b9829ccdb54c10dc3cb7a6f", hash);
+        } catch (final NamingServiceException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Test
+    public void namehash2() {
+        try {
+            final String hash = resolution.namehash("brad.crypto");
+            assertEquals("0x756e4e998dbffd803c21d23b06cd855cdc7a4b57706c95964a37e24b47c10fc9", hash);
+        } catch (final NamingServiceException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Test
+    public void wrongDomainNamehash() {
+        try {
+            resolution.namehash("unsupported");
+        } catch (final NamingServiceException e) {
+            assertEquals(NSExceptionCode.UnsupportedDomain, e.getCode());
+        }
+    }
+
+    @Test
+    public void addr() {
+        try {
+            String address = resolution.addr("homecakes.crypto", "eth");
+            assertEquals("0xe7474D07fD2FA286e7e0aa23cd107F8379085037", address);
+        } catch(Exception e) {
+            e.printStackTrace();
+            assertTrue(false);
+        }
+    }
+
+    @Test
+    public void wrongDomainAddr() {
+        try {
+            resolution.addr("unregistered.crypto", "eth");
+        } catch(NamingServiceException e) {
+            assertEquals(NSExceptionCode.UnregisteredDomain, e.getCode());
+        }
+    }
+
+    @Test
+    public void UnknownCurrency() {
+        try {
+            resolution.addr("brad.crypto", "unknown");
+        } catch(NamingServiceException e) {
+            assertEquals(NSExceptionCode.UnknownCurrency, e.getCode());
+        }
+    }
+
+    @Test
+    public void notConfiguredCurrency() {
+        try {
+            resolution.addr("brad.crypto", "dodge");
+        } catch(NamingServiceException e) {
+            assertEquals(NSExceptionCode.UnknownCurrency, e.getCode());
```

Comment:
You can not test exceptions raising this way: if no exception raised from a method, the test will not catch it up. See the techinique we used in JS version of the library.